### DISCUSSION TITLE
What do you think about Angular signals?

### DISCUSSION THREAD

**Discussion by fedeAlterio** [reactiveui/ReactiveUI#3760] [Ideas]
Hi!
Do you think could make sense to implement something similar to [angular signals](https://angular.io/guide/signals)?
They are really an intuitive and simple solution to handle state while still being integrated with reactive programming.
[Something like that](https://github.com/fedeAlterio/SignalsDotnet).

Thank you in advance.

**Comment by anaisbetts** [comment]
ReactiveUI is basically identical semantics to Signals, though with perhaps a little less nice semantics for effectively writing WhenAny's. What in particular do you see as missing?

**Reply by fedeAlterio** [reply]
I think there are some reasons. Consider this code:
`Signal.Computed(() => !string.IsNullOrWhiteSpace(Username.Value) && !string.IsNullOrWhiteSpace(Password.Value));`
Whit reactiveUI one could write something like:
`this.WhenAnyValue(x => x.UserName, x => x.Password).Select(x => ...).ToProperty()`
The main differences are:
- A computed signal automatically tracks other signals for changes (this is the game changer in my opinion)
- A computed signal dynamically adapts and listens only for the strictly needed signals. For example if the username is empty, there is no need for listen also for the password changes, This could be done in the WhenAnyValue with some reactiveX operators such as Switch() maybe, but its not worth it and it can become very messy very fast
- A computed signal could listen also for observableCollections changes. For example `Signal.Computed(() => !string.IsNullOrWhiteSpace(Username.Value) && !string.IsNullOrWhiteSpace(Password.Value)) || SomeCollection.Value.Any(x => x.SomeBool));`
